backend/app/modules/octant_rewards/general/service/pending.py

```diff
@@ -0,0 +1,103 @@
+        all_projects: list[str],
+        matched_rewards: int,
+    ) -> FinalizedProjectRewards:
+        ...
+
+
+@runtime_checkable
+class OctantMatchedRewards(Protocol):
+    def get_matched_rewards(self, context: Context) -> int:
+        ...
+
+
+class PendingOctantRewards(Model):
+    patrons_mode: UserPatronMode
+    user_rewards: UserRewards
+    project_rewards: ProjectRewards
+    octant_matched_rewards: OctantMatchedRewards
+
+    def get_octant_rewards(self, context: Context) -> OctantRewardsDTO:
+        pending_snapshot = database.pending_epoch_snapshot.get_by_epoch(
+            context.epoch_details.epoch_num
+        )
+        matched_rewards = self.octant_matched_rewards.get_matched_rewards(context)
+
+        return OctantRewardsDTO(
+            staking_proceeds=int(pending_snapshot.eth_proceeds),
+            locked_ratio=Decimal(pending_snapshot.locked_ratio),
+            total_effective_deposit=int(pending_snapshot.total_effective_deposit),
+            total_rewards=int(pending_snapshot.total_rewards),
+            individual_rewards=int(pending_snapshot.all_individual_rewards),
+            operational_cost=int(pending_snapshot.operational_cost),
+            community_fund=pending_snapshot.validated_community_fund,
+            ppf=pending_snapshot.validated_ppf,
+            matched_rewards=matched_rewards,
+            patrons_rewards=self.patrons_mode.get_patrons_rewards(context),
+            leftover=self.get_leftover(context, pending_snapshot, matched_rewards),
+        )
+
+    def get_matched_rewards(self, context: Context) -> int:
+        return self.octant_matched_rewards.get_matched_rewards(context)
+
+    def get_leverage(self, context: Context) -> float:
+        allocations_sum = database.allocations.get_alloc_sum_by_epoch(
+            context.epoch_details.epoch_num
+        )
+        matched_rewards = self.get_matched_rewards(context)
+
+        return calculate_leverage(matched_rewards, allocations_sum)
+
+    def get_leftover(
+        self,
+        context: Context,
+        pending_snapshot: PendingEpochSnapshot,
+        matched_rewards: int,
+    ) -> int:
+        allocations = database.allocations.get_all(context.epoch_details.epoch_num)
+        _, user_rewards = self.user_rewards.get_claimed_rewards(context)
+        project_rewards = self.project_rewards.get_finalized_project_rewards(
+            context, allocations, context.projects_details.projects, matched_rewards
+        )
+
+        return context.epoch_settings.octant_rewards.leftover.calculate_leftover(
+            LeftoverPayload(
+                staking_proceeds=int(pending_snapshot.eth_proceeds),
+                operational_cost=int(pending_snapshot.operational_cost),
+                community_fund=int(pending_snapshot.community_fund)
+                if pending_snapshot.community_fund
+                else None,
+                ppf=int(pending_snapshot.ppf) if pending_snapshot.ppf else None,
```

Comment:
there is a property method in PendingEpochSnapshot class for getting a proper value for ppf (```validated_ppf```)

Reply:
ok